### ISSUE TITLE
[Feature Request] Please detect and remove Windows usernames from paths (in logging)

### ISSUE THREAD

**Issue by dio-gh** [RPCS3/rpcs3#10174]
Currently, the emu logs contain full paths, meaning that on Windows, if the user has either their gamefiles or their RPCS3 folder located somewhere under their home folders, the username will land in the log (and possibly uploaded to here or to Discord).

This means that possibly personal data ends up in the logfiles, which is not very cool. Since this is something actually detectable and removeable (as opposed to any other piece of sensitive data that the users may mistakenly put into the path), I'd ask for this detection and anonymization to be done.

Furthermore, the user should be prompted in the first launch welcome message, that should the paths leading to their RPCS3 installation or their game files contain any sensitive information, they should look out for that (and that usernames are not included in this, on Windows).

**Comment by Asinin3** [comment]
Agree with censoring names in username folder, I actually suggested this back in 2018. Would be bad to see someone get Doxxed and could be more likely to happen to some popular Youtuber sharing a log or something. Don't agree with the warning message though, might put some people off and seems kinda pointless.

Would be nice to have another way of identifying a user without their username folder though e.g when we have people ban evading. Maybe someone has an idea for that, like hashing the name of the username folder. It would be some gibberish but still identifiable.

**Comment by elad335** [comment]
What I think is to replace full user path with a token which represents the user directory ike $user-directory/...

**Comment by arabek** [comment]
`$HOME` is the standard way to refer to user dir in all unix environments. Maybe use that as a universal mnemonic?

**Comment by MSuih** [comment]
The problem here is that we have file path printing in multiple locations across the codespace. If even one of those isn't handled properly (or someone adds a new one without carefully checking it) then paths will be exposed in log.

**Comment by arabek** [comment]
@MSuih In optimal case - file path logging would be handled by a path_logger function that'd sanitise all the output, and all places the code logs from would go trough that path_logger function to not leak the sensitive data. Of course, if someone carelessly adds a new function without using it - it'll expose stuff, but one can't pessimistically assume that.

**Comment by MiYakuGaming** [comment]
I agree with this, I don't want my future business information revealed.

**Comment by dio-gh** [comment]
Wasn't taken seriously. Closing.

**Comment by elad335** [comment]
I don't understand why closing.

**Comment by HerrHulaHoop** [comment]
Don't see anyone rejecting this feature request. Just that doing it properly requires effort and nobody has invested it yet.

**Comment by dio-gh** [comment]
I have gotten the wrong

**Comment by dio-gh** [comment]
I must have misinterpreted the responses then. Either way, I'm unsubscribing, please track it to completion on behalf of me then.

**Comment by digant73** [comment]
still a valid FR to consider for an implementation?

**Comment by elad335** [comment]
I think so.
